hbase-common/src/main/java/org/apache/hadoop/hbase/RawCell.java

```diff
@@ -64,4 +65,19 @@ public static void checkForTagsLength(int tagsLength) {
       throw new IllegalArgumentException("tagslength " + tagsLength + " > " + MAX_TAGS_LENGTH);
     }
   }
+
+  /**
+   * @return A new cell which is having the extra tags also added to it.
+   */
+  public static Cell createCell(Cell cell, List<Tag> tags) {
+    return PrivateCellUtil.createCell(cell, tags);
+  }
+
+  /**
+   * @param cell The Cell
+   * @return Tags in the given Cell as a List
+   */
+  public static List<Tag> getTags(Cell cell) {
```

Comment:
We already have Iterator<Tag> getTags() here right?  Thats good enough?

Reply:
We are exposing RawCell#createCell(Cell cell, List<Tag> tags) in this patch which accepts list of Tags.
So I thought it would be more user friendly to have some api that returns list of tags (instead of iterator), add a specific tag to the returned list and create a new cell with existing tags + 1 new tag. Open for suggestions.

Reply:
We have an instance method already here which return all Tags in this object.  So adding a static method here which similar behave seems a bit ugly for me.  I understood why u wanted to add this which helps to make the new Cell create code simpler there (Avoid going over iterator and collect tags to a List).  But I think its better if Phoenix (or other) do not try change the returned List by adding a new entry.  In many a places we return Immutable List impls. Here that is not needed for HBase but who knows may be at some time some patch might change that.  So that time Phoenix code will break. So i would suggest in phoenix code create a new List only and we can make use of the Iterator.

Reply:
@anoopsjohn makes sense.